Sprint_5

Silena812 opened the pull request.

Silena812 committed:
test_registration_PASSED

Silena812 committed:
tests for invalid password Passed

Silena812 committed:
reqistration with empty password test  passed

Silena812 committed:
enter button tests passed

Silena812 committed:
test go to account passed

Silena812 committed:
test go to constructor passed

Silena812 committed:
test exit passed

Silena812 committed:
test constructor sections passed

Silena812 committed:
all tests passed

Silena812 committed:
README added

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,37 @@
+from locators import Locators
+
+
+@pytest.fixture(scope="session")
```
Нужно исправить: для тестовых методов подобного типа драйвер должен запускаться для каждого отдельно

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,37 @@
+    service = Service()
+    browser = webdriver.Chrome(options=options, service=service)
+    browser.get(main_site)
+    print("Browser started")
```
Нужно исправить: для сообщений подобного рода print`ы избыточны. При желании можно использовать логгеры.

ShamilovR commented on curl.py:
```diff
@@ -0,0 +1,3 @@
+main_site = 'https://stellarburgers.nomoreparties.site/'
+login_site = 'https://stellarburgers.nomoreparties.site/login'
```
Можно лучше: урлы стоит конкатенировать с базовым, так будет проще поддерживать несколько стендов

ShamilovR commented on locators.py:
```diff
@@ -0,0 +1,33 @@
+
+class Locators:
+
+    ACCOUNT = (By.XPATH, "//*[@id='root']/div/header/nav/a/p") # Кнопка Личный Кабинет
```
Нужно исправить: здесь и далее: использование в локаторе порядкового номера элемента в dom, поиск от html, root, main и абсолютного пути порождает нестабильность локаторов (необходимо привязываться к атрибутам конкретного элемента)

ShamilovR commented on test_constructor.py:
```diff
@@ -0,0 +1,34 @@
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from locators import Locators
+
```
Нужно исправить: тестовые модули должны храниться в директории tests

ShamilovR commented on test_constructor.py:
```diff
@@ -0,0 +1,34 @@
+from selenium.webdriver.support import expected_conditions as EC
+from locators import Locators
+
+class TestConstructorSauces:
```
Нужно исправить: здесь и далее: тестовые классы используются для объединения тестовых методов в сьюты. Не нужно для каждого тестового метода выделать отдельный тестовый класс. Наоборот, в тестовом классе следует хранить набор тестовых методов, объединенных по логике.

Silena812 committed:
corrections: Нужно исправить: для тестовых методов подобного типа драйвер должен запускаться для каждого отдельно
Нужно исправить: для сообщений подобного рода print`ы избыточны. При желании можно использовать логгеры.
Нужно исправить: тестовые модули должны храниться в директории tests
Нужно исправить: здесь и далее: тестовые классы используются для объединения тестовых методов в сьюты.

Silena812 committed:
corrections: исправлены локаторы

Silena812 committed:
final commit
